pkg/reconciler/controllers/syncer.go

```diff
@@ -0,0 +1,343 @@
+			s.retryTimer.Reset(s.retryPeriod) // Schedule retry attempt
+			return ctrl.Result{}, nil
+		}
+		s.retry(trigger, false)
+	case parse.StatusUpdate:
+		klog.V(3).Info("Updating sync status (periodic while not syncing)")
+		// Skip sync status update if the .status.sync.commit is out of date.
+		// This avoids overwriting a newer Syncing condition with the status
+		// from an older commit.
+		if s.state.SyncCommitInStatus() == s.state.SourceCommitInStatus() &&
+			s.state.SyncCommitInStatus() == s.state.RenderingCommitInStatus() {
+			if err := parse.SetSyncStatus(s.controllerCtx, s.parser, s.state, s.parser.Syncing(), s.parser.SyncErrors()); err != nil {
+				klog.Warningf("failed to update sync status: %v", err)
+			}
+		}
+		s.statusUpdateTimer.Reset(s.statusUpdatePeriod) // Schedule status update attempt
+		s.retryTimer.Reset(s.retryPeriod)               // Schedule retry attempt
+	default:
+		klog.Errorf("unknown trigger %s", trigger)
+		s.statusUpdateTimer.Reset(s.statusUpdatePeriod) // Schedule status update attempt
+		s.retryTimer.Reset(s.retryPeriod)               // Schedule retry attempt
+	}
+
+	// No requeueing for the run failures because of the following reasons:
+	// 1. It only requeues run failures, but there are other failures that need to be
+	//    retried, e.g. management conflicts and updateWatch signal from the remediator.
+	// 2. the controller-runtime requeues the request using the default exponential
+	//    backoff, which is not tunable by Config Sync.
+	// 3. The default requeue can't filter out obsolete events.
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager registers the Syncer Controller.
+func (s *Syncer) SetupWithManager(mgr ctrl.Manager) error {
+	// an event channel to trigger reconciliation
+	timerEvents := make(chan event.GenericEvent)
+
+	// run the timer as a goroutine to trigger the reconciliation periodically
+	go func() {
+		// Use timers, not tickers.
+		// Tickers can cause memory leaks and continuous execution, when execution
+		// takes longer than the tick duration.
+		defer func() {
+			s.reimportTimer.Stop()
+			s.resyncTimer.Stop()
+			s.retryTimer.Stop()
+			s.statusUpdateTimer.Stop()
+		}()
+
+		s.state.InitRetryTimerAndPeriod(s.retryTimer, s.retryPeriod)
+		for {
+			select {
+			case <-s.controllerCtx.Done():
+				// The controller is shutting down, close the timerEvents channel and exit
+				close(timerEvents)
+				// Stop enqueueing and executing new syncing requests.
+				s.PauseSyncing()
+				// block and wait until the current syncing is done.
+				s.WaitSyncing()
+				close(s.doneCh) // inform finalizer the syncer is done
+				return
+
+			// Re-apply even if no changes have been detected.
+			// This case should be checked first since it resets the cache.
+			// If the reconciler is in the process of reconciling a given commit, the resync won't
+			// happen until the ongoing reconciliation is done.
+			case <-s.resyncTimer.C:
+				timerEvents <- timerEvent(parse.Resync)
```

Comment:
The old code reset the other timers after each event, but I don't see that code here. How is is delaying other events when a timer triggers?